File: Sources/BiangBiangUI/Protocols/Transliterator.swift
/// Converts an already-isolated run of script characters to Latin.
/// The library's `TextProcessingEngine` owns span detection, passthrough and spacing;
/// the app implementation only romanises one span. Must be pure and `Sendable`.
public protocol Transliterator: Sendable {
    func transliterate(_ scriptSpan: String) -> String
}

File: android/biangbiang-ui/src/main/java/dev/veeso/biangbiangui/protocols/Transliterator.kt
package dev.veeso.biangbiangui.protocols

/**
 * Converts an already-isolated run of script characters to Latin.
 * The library's text-processing engine owns span detection, passthrough and
 * spacing; the app implementation only romanises one span. Must be pure.
 * Mirrors iOS `Transliterator`.
 */
fun interface Transliterator {
    fun transliterate(scriptSpan: String): String
}
